File: node-agent/agentLlm.js
const { DASHSCOPE_BASE_URL, getEnv } = require("./envUtils");

class AgentLLM {
  constructor({ model } = {}) {
    this.model = model || getEnv("DASHSCOPE_MODEL", "qwen-plus");
    this.apiKey = getEnv("DASHSCOPE_API_KEY");
    this.baseUrl = getEnv("DASHSCOPE_BASE_URL", DASHSCOPE_BASE_URL);
  }

  async chat(prompt, history = [], metaInstruction = "") {
    if (!this.apiKey) {
      throw new Error("DASHSCOPE_API_KEY is not set");
    }

    const messages = [];
    if (metaInstruction) {
      messages.push({ role: "system", content: metaInstruction });
    }
    messages.push(...history);
    messages.push({ role: "user", content: prompt });

    const response = await fetch(
      `${this.baseUrl.replace(/\/$/, "")}/chat/completions`,
      {
        method: "POST",
        headers: {
          Authorization: `Bearer ${this.apiKey}`,
          "Content-Type": "application/json",
        },
        body: JSON.stringify({
          model: this.model,
          messages,
          temperature: 0.1,
        }),
      }
    );

    if (!response.ok) {
      throw new Error(`LLM request failed: ${response.status} ${await response.text()}`);
    }

    const data = await response.json();
    const aiResponse = data.choices[0].message.content;
    const newHistory = history.concat([
      { role: "user", content: prompt },
      { role: "assistant", content: aiResponse },
    ]);

    return [aiResponse, newHistory];
  }
}

module.exports = AgentLLM;

File: agent/agent_llm.py
#!/usr/bin/env python
# -*- coding: utf-8 -*-

'''
@File    :   agent_llm.py
@Time    :   2025/06/24 10:00:00
@Author  :   codemilestones
@Version :   1.0
@Desc    :   Agent LLM
'''
from typing import Optional

from env_utils import DASHSCOPE_BASE_URL, get_env


class AgentLLM:
    def __init__(self, path: str = '', model: Optional[str] = None) -> None:
        self.model = model or get_env("DASHSCOPE_MODEL", "qwen-plus")
        self.api_key = get_env("DASHSCOPE_API_KEY")
        self.base_url = get_env("DASHSCOPE_BASE_URL", DASHSCOPE_BASE_URL)

    def chat(self, prompt: str, history: list[dict], meta_instruction: str) -> tuple[str, list[dict]]:
        from openai import OpenAI

        if not self.api_key:
            raise RuntimeError("DASHSCOPE_API_KEY is not set")

        client = OpenAI(
            api_key=self.api_key,
            base_url=self.base_url,
        )

        messages = []
        if meta_instruction:
            messages.append({"role": "system", "content": meta_instruction})
        for entry in history:
            messages.append(entry)
        messages.append({"role": "user", "content": prompt})

        response = client.chat.completions.create(
            model=self.model,
            messages=messages,
            temperature=0.1
        )

        ai_response = response.choices[0].message.content
        new_history = history.copy()
        new_history.append({"role": "user", "content": prompt})
        new_history.append({"role": "assistant", "content": ai_response})

        return ai_response, new_history
